Data/binfmts/CMakeLists.txt

```
@@ -13,5 +13,25 @@ function(GenBinFmt Name)
     DESTINATION ${CMAKE_INSTALL_PREFIX}/share/binfmts/)
 endfunction()
 
-GenBinFmt(FEX-x86.in)
-GenBinFmt(FEX-x86_64.in)
+function(GenBinFmt_Systemd Name)
+  # Get the filename only component
+  get_filename_component(FMT_NAME ${Name} NAME_WE)
+
+  # Configure it
+  configure_file(
+    ${Name}
+    ${CMAKE_BINARY_DIR}/Data/binfmts/${FMT_NAME}.conf)
+
+  # Then install the configured binfmt
+  install(
+    FILES ${CMAKE_BINARY_DIR}/Data/binfmts/${FMT_NAME}.conf
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/binfmt.d/)
+endfunction()
+
+if (ENABLE_SYSTEMD_BINFMTMISC)
+  GenBinFmt_Systemd(FEX-x86.conf.in)
+  GenBinFmt_Systemd(FEX-x86_64.conf.in)
```

Inlining the function seems more readable overall since it's much shorter and doesn't require you to mentally substitute the function argument:
```suggestion
if (ENABLE_SYSTEMD_BINFMTMISC)
  configure_file(FEX-x86.conf.in ${CMAKE_BINARY_DIR}/Data/binfmts/FEX-x86.conf)
  configure_file(FEX-x86_64.conf.in ${CMAKE_BINARY_DIR}/Data/binfmts/FEX-x86_64.conf)

  install(
    FILES ${CMAKE_BINARY_DIR}/Data/binfmts/FEX-x86.conf ${CMAKE_BINARY_DIR}/Data/binfmts/FEX-x86_64.conf
    DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/binfmt.d/)
```

This actually applies to the legacy path as well already. Maybe we can additionally set a `BINFMT_CONFIG_PREFIX` variable conditionally to `.conf`/`` and use the same code for both?

---

Sure. inlined.